R/helpers.R

```diff
@@ -121,26 +125,45 @@ output_to_df <- function(output, model_arguments, compartments) {
 #'
 #' # get the epidemic size at the halfway point
 #' epidemic_size(data, stage = 0.5)
-epidemic_size <- function(data, stage = 1.0, by_group = TRUE) {
+epidemic_size <- function(data, stage = 1.0, by_group = TRUE, deaths = TRUE) {
   # input checking for data
   checkmate::assert_data_table(data)
   checkmate::assert_logical(by_group)
+  checkmate::assert_logical(deaths)
   checkmate::assert_number(stage, lower = 0.0, upper = 1.0, finite = TRUE)
 
   stopifnot(
     "No 'recovered' compartment found in `data`, check model compartments" =
       "recovered" %in% unique(data$compartment)
   )
+  # if deaths are requested to be counted, but no "dead" compartment exists
+  # throw a message
+  if (isTRUE(deaths) && (!"dead" %in% unique(data$compartment))) {
```

Comment:
```suggestion
  if (deaths && (!"dead" %in% unique(data$compartment))) {
```

Since `deaths` has been asserted to be a logical earlier, we can avoid the overhead associated with `isTRUE()`